Executor docs

rmlarose opened the pull request:
Description
-----------

Closes #995.

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-types` (from the root directory of the repository) and fix any [mypy](https://mypy.readthedocs.io/en/stable/) errors.

    2. Run `make check-style` and fix any [flake8](http://flake8.pycqa.org) errors.

    3. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html) autoformatter.

  For more information, check the [Mitiq style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

andreamari reviewed (commented):
Thanks @rmlarose, this guide is very clear! 

There is only one thing that it confused me: the function `compute_density_matrix` gives approximated results with strange errors and, only after looking at its code, I realized that it has some default amplitude damping noise. 

I would propose adding the noise explicitly, or mentioning in words that there is amplitude damping noise in the simulation of  `compute_density_matrix`, or setting the noise to zero. 
 
Another small thing that surprised me is that the `zne_value` is wrapped in a tuple instead of being a scalar:

`ZNE value: (0.01901578903198243+0j)`

But this is probably unrelated to this PR.

andreamari commented:
> `ZNE value: (0.01901578903198243+0j)`
> 
> But this is probably unrelated to this PR.

Actually this is probably just the way in which complex numbers are printed. If so can we try to reformat to avoid the parenthesis: e.g. something like `print(f"{zne_value :.4f}")`

andreamari commented on docs/source/guide/guide-executors.myst:
````diff
@@ -0,0 +1,181 @@
+```{code-cell} ipython3
+batched_executor = Executor(batch_compute_density_matrix, max_batch_size=10)
+
+zne_value = zne.execute_with_zne(cirq.Circuit(cirq.H.on(q) for _ in range(5)), batched_executor, obs)
````
```suggestion
zne_value = zne.execute_with_zne(cirq.Circuit(cirq.H.on(q) for _ in range(4)), batched_executor, obs)
```
To get simpler folded circuits (without approximations).

rmlarose commented:
> There is only one thing that it confused me: the function `compute_density_matrix` gives approximated results with strange errors and, only after looking at its code, I realized that it has some default amplitude damping noise.
> 
> I would propose adding the noise explicitly, or mentioning in words that there is amplitude damping noise in the simulation of `compute_density_matrix`, or setting the noise to zero.

Added a note that there is noise in the simulation. Good flag, hope it wasn't too confusing.

> Actually this is probably just the way in which complex numbers are printed. If so can we try to reformat to avoid the parenthesis: e.g. something like `print(f"{zne_value :g}")`

Added.
